controllers/helpers.go

```diff
@@ -256,17 +257,26 @@ func GetSubnetUUID(ctx context.Context, client *nutanixClientV3.Client, peUUID s
 		foundSubnetUUID = *subnetIntentResponse.Metadata.UUID
 	} else if subnetName != nil {
 		filter := getFilterForName(*subnetName)
-		subnetClientSideFilter := getSubnetClientSideFilter(peUUID)
-		responseSubnets, err := client.V3.ListAllSubnet(ctx, filter, subnetClientSideFilter)
+		// Not using additional filtering since we want to list overlay and vlan subnets
+		responseSubnets, err := client.V3.ListAllSubnet(ctx, filter, nil)
 		if err != nil {
 			return "", err
 		}
 		// Validate filtered Subnets
 		foundSubnets := make([]*nutanixClientV3.SubnetIntentResponse, 0)
 		for _, s := range responseSubnets.Entities {
 			subnetSpec := s.Spec
-			if *subnetSpec.Name == *subnetName && *subnetSpec.ClusterReference.UUID == peUUID {
-				foundSubnets = append(foundSubnets, s)
+			if *subnetSpec.Name == *subnetName && subnetSpec.Resources != nil && subnetSpec.Resources.SubnetType != nil {
+				switch *subnetSpec.Resources.SubnetType {
+				case subnetTypeOverlay:
+					// Overlay subnets are present on all PEs managed by PC.
+					foundSubnets = append(foundSubnets, s)
+				// By default check if the PE UUID matches if it is not an overlay subnet.
+				default:
+					if *subnetSpec.ClusterReference.UUID == peUUID {
+						foundSubnets = append(foundSubnets, s)
+					}
+				}
```

Comment:
given it's just two types we deal with atm, we should perhaps just use a `if` conditional here.

Reply:
Fixed